packages/react/README.md

```diff
@@ -15,3 +15,155 @@
 # @solana/react
 
 This package contains React hooks for building Solana apps.
+
+## Hooks
+
+### `useSignMessage(uiWalletAccount)`
+
+Given a `UiWalletAccount`, this hook returns a function you can call to sign a byte array.
+
+#### Arguments
+
+A config object with the following properties:
+
+-   `message`: A `Uint8Array` of bytes to sign
+
+#### Returns
+
+An object with the following properties:
+
+-   `signature`: The 64-byte Ed25519 signature as a `Uint8Array`
+-   `signedMessage`: The `Uint8Array` of bytes signed by the wallet. These bytes might differ from the input bytes if the wallet modified the message
```

Comment:
We added a `signatureType` field to this to be able to support non-Ed25519 signatures but no one uses it yet, and I think it's probably the wrong way to even try. A more appropriate way to do it would be a separate feature like `signMessageWithSecp256r1` or something.

Reply:
Yeah, I stripped out the `signatureType` because this is the `@solana/react` packages and I took a guess that we'd never have any other type of signatures, ever.

If this were the hook for `experimental:signMessage` then it would be in the returns.